Extensions/xinbox.js

```diff
@@ -1064,6 +1064,44 @@ XKit.extensions.xinbox = new Object({
 		document.body.appendChild(script);
 	},
 
+	do_post_colors: function(){
+		var colors = XKit.extensions.xinbox.get_blog_colors();
+		$(".posts .post[data-tumblelog]").not(":has(.xinbox-color-strip)").each(function(j,i){
+			var blog = i.dataset.tumblelog;
+
+			var n = '<div class="xinbox-color-strip" ' +
+				 'style="position: absolute; right: 0; top: 0; '+
+				 'border-top: 30px solid '+colors[blog]+'; border-left: 30px solid transparent;"'+
+				 'title="Submitted to '+blog+'"></div>';
+
+			$(i).append(n);
+		});
+	},
+
+	get_blog_colors: function() {
+		var current = XKit.storage.get("xinbox", "stable_blogs");
+		var new_blogs = XKit.tools.get_blogs();
+		if (!current) {
+			current = new_blogs;
+		} else {
+			current = current.split(";");
+			var set = new Set(current);
```

Comment:
Set()'s cross-browser compatibility seems pretty bad on Chrome (only available since chrome 38 when we technically support version since 27). A more standard approach like, `current.forEach(function(c) { set[c] = true; })` seems preferable.